protocol-units/bridge/shared/tests/blockchain_service.rs

```diff
@@ -0,0 +1,47 @@
+use bridge_shared::bridge_monitoring::BridgeContractInitiatorEvent;
+use bridge_shared::types::{
+	Amount, BridgeTransferDetails, BridgeTransferId, HashLock, InitiatorAddress, RecipientAddress,
+	TimeLock,
+};
+use bridge_shared::{blockchain_service::ContractEvent, bridge_contracts::BridgeContractInitiator};
+use futures::StreamExt;
+use std::task::{Context, Poll};
+
+mod shared;
+
+use shared::testing::mocks::MockBlockchainService;
+
+#[tokio::test]
+async fn test_bridge_transfer_initiated() {
+	let mut blockchain_service = MockBlockchainService::build();
+
+	blockchain_service
+		.initiator_contract
+		.with_next_bridge_transfer_id("transfer_id")
+		.initiate_bridge_transfer(
+			InitiatorAddress("initiator"),
+			RecipientAddress::from("recipient"),
+			HashLock("hash_lock"),
+			TimeLock(100),
+			Amount(1000),
+		)
+		.await
+		.expect("initiate_bridge_transfer failed");
+
+	let mut cx = Context::from_waker(futures::task::noop_waker_ref());
```

Comment:
Again, it seems like our own runtime might be overkill here. If we're doing this for some kind ordering, shouldn't the assumption be that the `BlockchainService` will (and in fact must) provide its own ordering.

Reply:
I'm unsure about your references to "ordering" and "runtime," Liam. Could you clarify what you mean? However, I'm not clear on how this relates to implementing a runtime. There is no runtime implementation here. Only the check that items get cleaned up as they should. Could you please explain further?

Reply:
You're implementing poll and waker logic at a lower level than simply relying on async methods. Why?

Reply:
If you are aiming for a higher degree of control beyond the use of async methods, particularly for detailed testing, it becomes necessary to adopt a more nuanced implementation approach. 

This subject might warrant a thorough, dedicated discussion on design tradeoffs beyond the scope of a pull request.